imgur.sh

```diff
@@ -45,15 +45,22 @@ while [ $# -gt 0 ]; do
 	file="$1"
 	shift
 
-	# Check file exists
-	if [ "$file" != "-" -a ! -f "$file" ]; then
-		echo "File '$file' doesn't exist; skipping" >&2
-		errors=true
-		continue
-	fi
-
 	# Upload the image
-	response=$(curl -s -H "Authorization: Client-ID $client_id" -H "Expect: " -F "image=@$file" https://api.imgur.com/3/image.xml) 2>/dev/null
+	# URL -> imgur
+	if [[ "$file" =~ ^https?://* ]]; then
```

Comment:
That `*` shouldn't be there. This will currently match `http:/example`, since the last `/*` means "match 0 or more `/` characters".

And are there other protocols imgur supports downloading from, or is it just HTTP/HTTPS?